### DISCUSSION TITLE
Add recipe to use dataloader with gin

### DISCUSSION THREAD

**Discussion by mirusky** [99designs/gqlgen#1924] [Q&A] [answered]
### What happened?

I'm lost in how to implement dataloader and gin together

### What did you expect?

To avoid N+1 case
![image](https://user-images.githubusercontent.com/54698990/104679862-5b00de00-56cd-11eb-845f-9084ef91f0ba.png)

### Minimal graphql.schema and models to reproduce

The getting-started schema + database in resolver.go

### versions
 - `gqlgen version`? v0.13.0
 - `go version`? 1.15
 - dep or go modules? modules

**Comment by calam1** [comment, marked as answer]
So I hope this helps.  I battled through this like 10 months or so ago.  But I figured it out w/a little help from this one particular youtube vid - https://www.youtube.com/watch?v=D3SrjuTaUQU

I also added some snippets of code in a gist, hopefully that will add a little context.  But watch the video first, then I think the code snippets will make more sense.  It's been awhile since I have touched this code, so I don't remember everything.

https://gist.github.com/calam1/19cbd3ceef2d903f99a3c1696b94bdc1

**Comment by mirusky** [comment]
Ohh thanks @calam1, this video and the snippets helped me a lot to make some progress here. 

I will try to create a sample or a an example and make a pull request.
